MinGW builds fail with `invalid conversion from 'std::thread::native_handle_type' {aka 'unsigned int'} to 'HANDLE' {aka 'void*'}`

**Describe the bug**
Building vsg with MinGW gcc fails with the mentioned error

**To Reproduce**
This issue has been detected at https://build.opensuse.org/package/show/home:rhabacker:branches:games:mingw32/mingw32-libvsg

**Expected behavior**
The build should not fail

> Did you copy and past the de

Thanks for pointer, I updated the description.

> Could you just post the errors directly rather than this other website?
```
[  125s] /home/abuild/rpmbuild/BUILD/VulkanSceneGraph-0.1.4.20210909.git+7847d18a/src/vsg/threading/Affinity.cpp: In function 'void vsg::setAffinity(std::thread&, const vsg::Affinity&)':
[  125s] /home/abuild/rpmbuild/BUILD/VulkanSceneGraph-0.1.4.20210909.git+7847d18a/src/vsg/threading/Affinity.cpp:49:43: error: invalid conversion from 'std::thread::native_handle_type' {aka 'unsigned int'} to 'HANDLE' {aka 'void*'} [-fpermissive]
[  125s]    49 |     win32_setAffinity(thread.native_handle(), affinity);
[  125s]       |                       ~~~~~~~~~~~~~~~~~~~~^~
[  125s]       |                                           |
[  125s]       |                                           std::thread::native_handle_type {aka unsigned int}
[  125s] /home/abuild/rpmbuild/BUILD/VulkanSceneGraph-0.1.4.20210909.git+7847d18a/src/vsg/threading/Affinity.cpp:20:38: note:   initializing argument 1 of 'void win32_setAffinity(HANDLE, const vsg::Affinity&)'
[  125s]    20 | static void win32_setAffinity(HANDLE tid, const vsg::Affinity& affinity)
[  125s]       |                               ~~~~~~~^~~
[  125s] make[2]: *** [src/vsg/CMakeFiles/vsg.dir/build.make:1498: src/vsg/CMakeFiles/vsg.dir/threading/Affinity.cpp.obj] Error 1
```

--- a/src/vsg/threading/Affinity.cpp
+++ b/src/vsg/threading/Affinity.cpp
@@ -46,7 +46,7 @@ static void win32_setAffinity(HANDLE tid, const vsg::Affinity& affinity)
 
 void vsg::setAffinity(std::thread& thread, const Affinity& affinity)
 {
-    win32_setAffinity(thread.native_handle(), affinity);
+    win32_setAffinity((HANDLE)thread.native_handle(), affinity);
 }
 
 void vsg::setAffinity(const Affinity& affinity)

